crates/red_knot_python_semantic/src/semantic_index/use_def/symbol_state.rs

```diff
@@ -173,11 +171,46 @@ impl SymbolDeclarations {
             visibility_constraints: self.visibility_constraints.iter(),
         }
     }
+
+    fn merge(&mut self, b: Self, visibility_constraints: &mut VisibilityConstraints) {
+        let mut a = Self::default();
+        std::mem::swap(&mut a, self);
+        self.live_declarations = a.live_declarations.clone();
+        self.live_declarations.union(&b.live_declarations);
+        let a = izip!(
+            a.live_declarations.iter(),
+            a.visibility_constraints.into_iter()
+        );
+        let b = izip!(
+            b.live_declarations.iter(),
+            b.visibility_constraints.into_iter()
+        );
```

Comment:
I'd use the regular `Iterator::zip` method here because it's better known and we don't use the functionality that `izip` provides

> This is a version of the standard .zip() that’s supporting more than two iterators. 
```suggestion
        let a = a.live_declarations.iter().zip(a.visibility_constraints);
        let b = b
            .live_declarations
            .iter()
            .zip(b.visibility_constraints.into_iter());
```